### PR TITLE
constexpr char literal for cigar

### LINKED ISSUE #2819: Constexpr for the CIGAR char literal
In some contexts, the cigar character needs to be used in constexpr context, e.g.
```cpp
    seqan3::cigar::operation op = 'M'_cigar_operation;
    switch (op.to_rank())
    {
        case 'M'_cigar_operation.to_rank():
            ;
    }
```

I think that it is useful to implement this and I'll provide a PR for it.

### PR DESCRIPTION
**Pull request by joergi-w**
Resolves #2819

### CODE REVIEW

**Review by MitraDarja — APPROVED** [review]

File: test/unit/alphabet/cigar/cigar_test.cpp [hunk 1]
```diff
@@ -56,3 +56,16 @@ TEST(cigar, assign_string)
     EXPECT_EQ(uint32_t{223}, seqan3::to_rank(get<0>(c1)));
     EXPECT_EQ('M',           get<1>(c1).to_char());
 }
+
+TEST(cigar, constexpr_char_literal)
+{
+    using seqan3::operator""_cigar_operation;
+    seqan3::cigar::operation op = 'M'_cigar_operation;
```
Comment (eseiler) [inline test/unit/alphabet/cigar/cigar_test.cpp:63]:
```suggestion
    constexpr seqan3::cigar::operation op = 'M'_cigar_operation;
```

**Comment by eseiler** [comment]
@marehr Should go into release branch?

File: test/unit/alphabet/cigar/cigar_test.cpp (hunk 1 above)
Comment (marehr) [reply test/unit/alphabet/cigar/cigar_test.cpp:63]: I would change from 'M' as it is rank == 0; to 'D' which has rank 1.

```suggestion
    constexpr seqan3::cigar::operation op = 'D'_cigar_operation;
```

File: test/unit/alphabet/cigar/cigar_test.cpp [hunk 2]
```diff
@@ -56,3 +56,16 @@ TEST(cigar, assign_string)
...
+    switch (op.to_rank())
+    {
+        case 'M'_cigar_operation.to_rank():
+            break;
+        default:
+            ;
+    }
```
Comment (marehr) [inline test/unit/alphabet/cigar/cigar_test.cpp:70]: After the change from @eseiler the whole switch statement doesn't add anything. 

```suggestion
    EXPECT_EQ(op.to_rank(), 1u);
```

**Comment by marehr** [comment]
@eseiler Yes, seems like a good idea. Change is small enough :)

File: include/seqan3/alphabet/cigar/cigar.hpp [hunk 3]
```diff
@@ -235,7 +235,7 @@ inline namespace literals
  *
  * \stableapi{Since version 3.1.}
  */
-inline cigar::operation operator""_cigar_operation(char const c) noexcept
+inline constexpr cigar::operation operator""_cigar_operation(char const c) noexcept
```
Comment (marehr) [inline include/seqan3/alphabet/cigar/cigar.hpp:238]:
```suggestion
constexpr cigar::operation operator""_cigar_operation(char const c) noexcept
```

**Comment by marehr** [comment]
@joergi-w Can you rebase in such a way that the PR goes against `upstream/release-3.1.0` ?

**Comment by joergi-w** [comment]
Thank you for the helpful review comments! The switch statement in the test was my use case to detect the issue, but your proposal is surely shorter to test `constexpr`-ness. 

I have rebased on release-3.1.0 and checked that all the other alphabets have constexpr char literal as well.

**Review by marehr — APPROVED** [review]

**Pull request merged**